docs/proposals/docker/docker_image_module.md

```diff
@@ -0,0 +1,201 @@
+
+Docker_image will manage images using docker-py to communicate with either a local or remote API. It will
+support API versions >= 1.14. API connection details and registry authentication will be handled externally in a
+shared utility module similar to how other cloud modules operate.
+
+## Parameters
+
+Docker_image will support the parameters listed below. API connection parameters will be part of a shared utility module
+as mentioned above.
+
+```
+archive_path:
+  description:
+    - File name for the archive when saving an image.
+  default: null
+
+build_args:
+  description:
+    - List of build-time variables.
+  default: null
+
+cgroup_parent:
+  description:
+    - Optional parent cgroup for build containers.
+  default: null
+
+cpu_period:
+  description:
+    - Limit build container CPU CFS (Completely Fair Scheduler) period. Expressed in milliseconds.
+  default: 0
+
+cpu_quota:
+  description:
+    - Limit build container CPU CFS (Completely Fair Scheduler) quota. Expressed in milliseconds.
+  default: 0
+
+cpu_shares:
+  description:
+    - CPU shares for build containers. Integer value.
+  default: 0
+
+cpuset_cpus:
+  description:
+    - CPUs in which to allow build container execution C(1,3) or C(1-3).
+  default: null
+
+cpuset_mems:
+  description:
+    - Memory nodes (MEMs) in which to allow build container execution. For example, C(0-3) or C(0,1).
+  default: null
+
+dockerfile:
+  description:
+    - Name of dockerfile to use when building an image.
+  default: Dockerfile
+
+force:
+  description:
+    - Use with absent state to untag and remove all images matching the specified name. Use with built state to
+      force rebuilding of an image.
+  default: false
+
+name:
+  description:
+    - Image name or Image ID.
+  required: true
+
+nocache:
+  description:
+    - Do not use cache when building
+  deafult: false
+
+path:
+  description:
+    - Path to Dockerfile and context from which to build an image.
+  default: null
+
+registry:
+  description:
+    - host or host:port of a remote registry.
+  required: false
+
+rm:
+  description:
+    - Remove itermediate containers after build.
+  default: true
+
+shm_size:
+  description:
+    - For build containers. Size of `/dev/shm`. The format is `<number><unit>`. `number` must be greater than `0`.
+      Unit is optional and can be `b` (bytes), `k` (kilobytes), `m` (megabytes), or `g` (gigabytes).
+    - Ommitting the unit defaults to bytes. If you omit the size entirely, the system uses `64m`.
+  default: null
+
+tag:
+  description:
+    - Image tags. When pulling or pushing, set to 'all' to include all tags.
+  default: "latest
+
+ulimit:
+  description:
+    - List of ulimit options passed to build containers. A ulimit is specified as C(nofile=262144:262144)
+  default: null
+
+url:
+  description:
+    - The location of a Git repository. The repository acts as the context when building an image.
+    - Mutually exclusive with path.
+
+state:
+  description:
```

Comment:
many of these states are not needed and can just be inferred from the other options, for example 'saved' should happen when 'archive_path' is not null.